packages/core/svelte/src/walletStore.ts

```diff
@@ -256,7 +259,9 @@ function onDisconnect() {
 
 walletNameStore.subscribe(({ walletName }: { walletName: WalletName | null }) => {
     const { localStorageKey } = get(walletConfigStore);
-    setLocalStorage(localStorageKey, walletName);
+    if (walletName){
+        setLocalStorage(localStorageKey, walletName);
+    }
```

Comment:
We discussed this before, and I'm still not sure about this change. The idea here is that if `walletName` is `null` (which is explicitly allowed as an arg), then `setLocalStorage` should still be called with `null` in order to remove the item. This is what should be done any time the wallet is imperatively disconnected.

Reply:
Yes, we discussed this before and after you comment, I went to check it out again and I see what is happening. 

#### Doing your solution:

The subscribe method is listening since the very first value of `walletName`, and initially this is `null`. That's why autoConnect was not working: as soon I refreshed, the value was `null` and the localStorage removed. 

#### Doing my solution:
It didn't work either because there is not chance to remove the localStorage when we disconnect the wallet and the `walletName` is null. 

#### a new proposal
I've tested this several times and it is working as expected. What I want is that `setLocalStorage(localStorageKey, walletName);` is going to work when `initialize` is called, and the only way to know it is when `wallets.length > 0`. In this case we can check the changing value of `walletName` and save or remove the localStorage. 

```
if (wallets.length > 0) {
    setLocalStorage(localStorageKey, walletName);
}
```

Reply:
Hmm. This third proposal looks much more hacky than I'm comfortable with. I think we need to go back to the drawing board and take cues from how the React and Vue packages handle more of the implementation.

Reply:
@jordansexton investigating a bit further about this, the `subscribe` method reads the initial Store value and the changes related with it. The same that reactivity `$` in the Svelte components. In this case we need to ensure we are triggering `setLocalStorage` after the wallet is initialized. The wallets array in the only parameter which can ensure that it is happening.

Reply:
Counting on the wallets array not to be populated doesn't feel like a future proof design (it's merely incidental to what the intent of this code is, which is to control localStorage).

What I meant above is that the React and Vue implementations are more straightforward to follow and maintain because the side effects of the code are controlled. We should try to follow them and bring this approach to the Svelte implementation, which right now is much less easy to follow logically, and there are several behaviors that are occurring that we don't have a complete understanding of yet.

Reply:
There is another solution and is to call directly the function like this: 

on the `select()` function we can do `setLocalStorage(localStorageKey, newName);`
and in the `disconnect()`, `setLocalStorage(localStorageKey, null);`

The effect is the same and we prevent to call the function at the first load of the subscribe method. I think the Vue implementation the the cleanest, but Svelte ( React neither ) has not the Composition API. It is impossible to have the same solution for the 4 frameworks, since Svelte is the only one pure reactive. And the structure of the Svelte implementation is clear. We are used to work with many Stores. Basically to build this implementation, I brought the Vue one line by line and adapt it to the Svelte logic. 

Even we are making something totally isolated and not depending of Svelte Components as in the first commits of the first version. The React one needs the WalletProvider.ts for example. What I mean is that, each framework has it own logic and regarding complexity, the Angular impl is not very straightforward.

Reply:
> Counting on the wallets array not to be populated doesn't feel like a future proof design (it's merely incidental to what the intent of this code is, which is to control localStorage).
> 
> What I meant above is that the React and Vue implementations are more straightforward to follow and maintain because the side effects of the code are controlled. We should try to follow them and bring this approach to the Svelte implementation, which right now is much less easy to follow logically, and there are several behaviors that are occurring that we don't have a complete understanding of yet.

Just to chime in; I haven't followed this implementation or the react / vue implementations closely, but following this thread, this particular line of code is guarding against the standard behaviour of a svelte store. The moment you `.subsribe` on a store, the current values in that store are emitted. 

On an initial load, those vales will be `null` which will trigger the removal of the `localStorage` item (incorrectly). We could add a safeguard around that to disregard the first value emitted, which will solve this problem. Personally I think adding the safeguard is OK for now, as this is just how svelte stores behave naturally.

Some more general points;
1. It will be nice to write idiomatic svelte code, rather than mirror (which I don't think is what you're suggesting) the react / vue implementation
2. It's more important to get the overall API design of the core package correct. As its easier to fix/change implementation than to change the API once it's actually in use. The API seems to be inline with the other implementations

@silvestrevivo are my findings correct?

Reply:
@cogoo Your explanation is very accurate and essentially that is what is happening. Regarding your points:

1. I think the solution is aligned with the Svelte style. It is true that is mirroring the Vue implementation since Vue and Svelte are quite similar. But as we have mentioned before, in the same way that the React iml has a component like `WalletProvider` or `WalletConnection` as part of the implementation, we could do something similar for this one. But in this case, as you requested in the previous PR, he have worked hard to solve it just in one file and in an isolated way. Like the Vue implementation. 
2. I think as well the API is aligned with the others.